lua/neo-tree/sources/common/components.lua

```diff
@@ -488,7 +495,7 @@ M.file_size = function (config, node, state)
   end
 
   return {
-    text = string.format("%12s  ", text),
+    text = vim.fn.printf("%" .. config.width .. "s  ", truncate_string(text, config.width)),
```

Comment:
If the string is truncated to the correct length anyways, maybe simplify this to:

```suggestion
    text = truncate_string(text, config.width) .. "  ",
```